src/test/util/random.h

```diff
@@ -25,7 +25,7 @@ enum class SeedRand {
 };
 
 /** Seed the RNG for testing. This affects all randomness, except GetStrongRandBytes(). */
-void SeedRandomForTest(SeedRand seed = SeedRand::SEED);
+void SeedRandomForTest(FastRandomContext& rng, SeedRand seed = SeedRand::SEED);
```

Comment:
> nit in [bb67ca0](https://github.com/bitcoin/bitcoin/commit/bb67ca09c88a46b99bf83e937fa761f0e25cba37): Not sure about this change. It seems overly restrictive to require a `FastRandomContext&` here, when a caller may not want to pass anything at all, or a different rng.

Makes sense. I hadn't really looked at this function closely and was just trying to keep behavior unchanged. Maybe it would be good to change signature to something like:

```c++
SeedRandomForTest(SeedRand seed = SeedRand::SEED, FastRandomContext* rng = nullptr)
```

and change the last line of the function to

```c++
if (rng) rng->Reseed(GetRandHash());
```

so it could still be used conveniently to seed m_rng. Or maybe the parameter could just be dropped like the global `g_insecure_rand_ctx` is dropped, and `m_rng` could be managed separately.